requests_mv_integrations/request_mv_integration.py

```diff
@@ -0,0 +1,1774 @@
+
+            self.logger.info(
+                "Request Retry: Performing Retry",
+                extra={
+                    'tries': _tries,
+                    'delay': _delay,
+                    'timeout': _timeout,
+                    'request_url': request_url,
+                    'request_label': request_label
+                }
+            )
+
+            time.sleep(_delay)
+
+            if retry_backoff and retry_backoff > 0:
+                _delay *= retry_backoff
+
+            if retry_jitter and retry_jitter > 0:
+                _delay += retry_jitter
+
+            if retry_max_delay is not None:
+                _delay = min(_delay, retry_max_delay)
+
+    # Request Data
+    #
+    def _request(
+        self,
+        request_method,
+        request_url,
+        request_params=None,
+        request_data=None,
+        request_json=None,
+        request_headers=None,
+        request_auth=None,
+        request_session=False,
+        cookie_payload=None,
+        request_label=None,
+        timeout=60,
+        build_request_curl=True,
+        allow_redirects=True,
+        verify=True,
+        stream=False,
+        verbose=False
+    ):
+        """Constructs and sends a :class:`Request <Request>`.
+
+        Args:
+            request_method: request_method for the new :class:`Request` object.
+            logger: logging instance
+            request_url: URL for the new :class:`Request` object.
+            request_params: (optional) Dictionary or bytes to be sent in the
+                query string for the :class:`Request`.
+            request_data: (optional) Dictionary, bytes, or file-like object to
+                send in the body of the :class:`Request`.
+            request_json: (optional) json data to send in the body of
+                the :class:`Request`.
+            request_headers: (optional) Dictionary of HTTP Headers to send
+                with the :class:`Request`.
+            request_auth: (optional) Auth tuple to enable Basic/Digest/Custom HTTP Auth.
+            timeout: (optional) How long to wait for the server to send data
+                before giving up.
+            allow_redirects: (optional) Boolean. Set to True if POST/PUT/DELETE
+                redirect following is allowed.
+            verify: (optional) whether the SSL cert will be verified. A
+                CA_BUNDLE path can also be provided. Defaults to ``True``.
+            stream: (optional) if ``False``, the response content will be
+                immediately downloaded.
+            verbose: (optional) Boolean, provide request and response details.
+
+        Returns:
+            requests.Response
+
+        """
+        tune_request = TuneRequest(
+            retry_tries=self.retry_tries,
+            retry_backoff=self.retry_backoff,
+            retry_codes=self.request_retry_http_status_codes
+        )
+
+        self.session = tune_request.session
+
+        self.logger.debug(
```

Comment:
I'm not favorable towards these logs that print all of the arguments I just passed to the function. This is something a debugger/tracer is built to handle for you and they are not useful while building an integration. We can also write one off debug statements when we need them, but theres no reason for them to live in our code.